src/Config/Database/DbConfig.php

```diff
@@ -0,0 +1,299 @@
+
+        if (!$this->configMerger->isEmpty($customConfig) && !$this->configMerger->isMergeRequired($customConfig)) {
+            return $this->configMerger->clear($customConfig);
+        }
+
+        return $this->dbConfig = $this->configMerger->merge($this->collectEnvConfig($customConfig), $customConfig);
+    }
+
+    /**
+     *  Collects DB connections which exist in environment.
+     *
+     * @param array $customDbConfig
+     * @return array
+     */
+    private function collectEnvConfig(array $customDbConfig): array
+    {
+        $config = [];
+        foreach (self::MAIN_CONNECTION_MAP as $mageConnectionName => $envConnectionName) {
+            // collect main connections
+            $envDbConfig = $this->getEnvConnectionData($envConnectionName);
+            if (empty($envDbConfig->getHost())) {
+                continue;
+            }
+            $config[self::KEY_CONNECTION][$mageConnectionName] = $this->convertToMageFormat(
+                $envDbConfig,
+                !in_array($mageConnectionName, self::MAIN_CONNECTIONS)
+            );
+            // collect slave connections
+            if (!isset(self::SLAVE_CONNECTION_MAP[$mageConnectionName])) {
+                continue;
+            }
+            $slaveConnectionData = $this->getEnvConnectionData(self::SLAVE_CONNECTION_MAP[$mageConnectionName]);
+            if (empty($slaveConnectionData->getHost())
+                || !$this->isCustomConnectionCompatibleForSlave($customDbConfig, $mageConnectionName, $envDbConfig)) {
+                continue;
+            }
+            $mageConnectionConfig = $this->convertToMageFormat($slaveConnectionData, true);
+            $config[self::KEY_SLAVE_CONNECTION][$mageConnectionName] = $mageConnectionConfig;
+        }
+        return $config;
+    }
+
+    /**
+     * Returns a custom database configuration from the variable DATABASE_CONFIGURATION from .magento.env.yaml
+     * Custom split connections are removed as they are not supported.
+     *
+     * @return array
+     */
+    private function getSupportedCustomConfig(): array
+    {
+        $customConfig = (array)$this->stageConfig->get(DeployInterface::VAR_DATABASE_CONFIGURATION);
+
+        /**
+         * Ece-tools do not support custom configuration of a split database.
+         */
+        foreach (self::SPLIT_CONNECTIONS as $splitConnection) {
+            foreach (self::CONNECTION_TYPES as $connectionType) {
+                if (isset($customConfig[$connectionType][$splitConnection])) {
+                    unset($customConfig[$connectionType][$splitConnection]);
+                }
+            }
+        }
+        return $customConfig;
+    }
+
+    /**
+     * Checks that database configuration was changed in DATABASE_CONFIGURATION variable
+     * in not compatible way with slave_connection.
+     *
+     * Returns true if $envDbConfig contains host or dbname for default connection
+     * that doesn't match connection from relationships,
```

Comment:
Seems like there should be `does match` instead of `doesn't`

Reply:
Done